### DISCUSSION TITLE
library chunks not sharing the same react context instance

### DISCUSSION THREAD

**Discussion by ghost** [webpack/webpack#17289] [Q&A]
.

**Comment by alexander-akait** [comment]
When you build libraries you don't need `splitChunks`. Setup `react` and react related libraries as external

**Reply by alexander-akait** [reply]
So now you have less size, because you don't have extra `react` library, also developer can use any supported react version with your library

**Comment by ghost** [comment]


**Reply by alexander-akait** [reply]
> sadly this is not possible with webpack.

What is not true, sometimes for rollup you need just less options, but almost everything can be builded with webpack

> Usually what you should do is extract the shared part into a separate bundle, and then use externals to prevent it from being copied into other bundles.

You don't need multipl entrypoints for library (except when you know what you do), also you don't need `dependOn` (again if you don't know what you do). 

Sorry I can't undestand your context and your problem, as I already say, everything related to react should be `externals`, otherwise you will have bloated library
